### ISSUE TITLE
Pathlib: Conversion between PureWindowsPath and PurePosixPath mangles root.

### ISSUE THREAD

**Issue by ghost** [python/cpython#115039]
# Bug report

### Bug description:

When PureWindowsPath gets converted to PurePosixPath
```python
>>> from pathlib import PureWindowsPath, PurePosixPath
>>> a = PureWindowsPath("/hello/world/2/")
>>> PurePosixPath(a)
PurePosixPath('\\/hello/world/2')
>>>
```
the root doesn't get correctly converted, this should convert
to /hello/world/2, as "\\" means root of current virtual disk.

### CPython versions tested on:

3.11

### Operating systems tested on:

Linux, Windows

**Comment by barneygale** [comment]
Should be fixed in 3.12 - can you test there?

**Comment by deleteData759** [comment]
> Should be fixed in 3.12 - can you test there?
Sure, glad to recompile Python after a while.
Used to be a contributor and I'm looking for some ways to come back.

Any fun bugs around importlib, built-in modules?

**Comment by deleteData759** [comment]
> Should be fixed in 3.12 - can you test there?

Tested on main. I hate you guys for main, but I understand.
It's fixed there.
Testing 3.12... (I'm writing this comment as I'm testing it.)
behaves as expected.

Time to update Python I guess.

**Comment by deleteData759** [comment]
Closing comment because M*oft demands so.
